trailblazer/apps/slurm/api.py

```diff
@@ -0,0 +1,62 @@
+from typing import List, Dict, Optional
+
+from trailblazer.apps.slurm.models import SqueueResult
+from trailblazer.constants import FileFormat, SlurmJobStatus, TrailblazerStatus
+from trailblazer.exc import EmptySqueueError
+from trailblazer.io.controller import ReadStream
+from trailblazer.apps.slurm.utils import formatters
+
+
+def get_squeue_result(squeue_response: str) -> SqueueResult:
+    """Return SqueueResult object from squeue response.
+    Raises:
+        TrailblazerError: when no entries were returned by squeue command.
+    """
+    if not squeue_response:
+        raise EmptySqueueError("No jobs found in SLURM registry")
+    squeue_response_content: List[dict] = ReadStream.get_content_from_stream(
+        file_format=FileFormat.CSV,
+        stream=squeue_response,
+        read_to_dict=True,
+    )
+    return SqueueResult(jobs=squeue_response_content)
+
+
+def reformat_squeue_result_job_step(data_analysis: str, job_step: str) -> str:
+    """Standardise job step string according to data anlaysis."""
+    formatter_func = formatters.formatter_map.get(data_analysis, formatters.reformat_undefined)
+    return formatter_func(job_step=job_step)
+
+
+def get_current_analysis_status(jobs_status_distribution: dict) -> Optional[str]:
+    """Return current analysis status based on jobs status distribution."""
+    is_ongoing: bool = bool(
+        jobs_status_distribution.get(SlurmJobStatus.RUNNING)
+        or jobs_status_distribution.get(SlurmJobStatus.PENDING)
+    )
+
+    analysis_status_map: Dict[str, Optional[str]] = {
```

Comment:
can this be extracted into a class that handles the logic in smaller functions, this is difficult to follow.

Reply:
In this case I do not think it will help readability as the functions will take 3 or 2 parameters and return the same line as is in the dict already. This will make it harder to see the overall logic and just make it more abstract without making it clearer.

Reply:
I see, makes sense to keep it that way then.